build.rs

```diff
@@ -84,6 +86,26 @@ fn main() {
     if linux_time_bits64 {
         set_cfg("linux_time_bits64");
     }
+    println!("cargo:rerun-if-env-changed=RUST_LIBC_UNSTABLE_GNU_FILE_OFFSET_BITS64");
+    match env::var("RUST_LIBC_UNSTABLE_GNU_FILE_OFFSET_BITS64") {
+        Ok(val) => {
+            if val == "1" {
+                let target_env = env::var("CARGO_CFG_TARGET_ENV").unwrap();
+                let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
+                let target_ptr_width = env::var("CARGO_CFG_TARGET_POINTER_WIDTH").unwrap();
+                let target_arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap();
```

Comment:
Nit, could you move these out of this block and toward the top of `main`? Something else may want to use them.

`unwrap_or_default()` may be better too, I doubt anybody is building this without Cargo but it's probably good to be careful just in case.